### PR TITLE
Update the rust submodule we're testing

### PR DESCRIPTION
**Pull request by alexcrichton**
Was playing around with async/await and ran into syntax that syn can't parse. Took at stab in the dark an prayed that updating the `rust` submodule and fixing tests would fix my original test. It didn't. Figured this'd be useful regardless!

### CODE REVIEW

File: tests/test_precedence.rs [hunk 1]
```diff
@@ -91,6 +89,9 @@ fn test_rustc_precedence() {
         if path.is_dir() {
             continue;
         }
+        if path.to_str().unwrap().ends_with("optional_comma_in_match_arm.rs") {
```
Comment (mystor) [inline tests/test_precedence.rs]: nit: Can you add a comment here explaining why this test fails?

File: src/main.rs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+extern crate syn;
+
+use std::io::{self, Read};
+
+fn main() {
+    // let mut input = String::new();
+    // io::stdin().read_to_string(&mut input).unwrap();
+    //
+    // syn::parse_str::<syn::File>(&input).unwrap();
```
Comment (mystor) [inline src/main.rs]: Please remove this file, I don't think it's necessary.

File: src/expr.rs [hunk 3]
```diff
@@ -624,11 +624,20 @@ ast_enum! {
 
 #[cfg(any(feature = "parsing", feature = "printing"))]
 #[cfg(feature = "full")]
-fn arm_requires_comma(arm: &Arm) -> bool {
-    if let ExprKind::Block(ExprBlock { unsafety: Unsafety::Normal, .. }) = arm.body.node {
-        false
-    } else {
-        true
+fn arm_expr_requires_comma(expr: &Expr) -> bool {
```
Comment (dtolnay) [inline src/expr.rs]: I don't think this is correct -- the rule for expressions that need a semicolon in order to be a statement is different from the rule for expressions that need a comma in order to be a match arm. The link is for the former. I haven't tried it but does this change make syn erroneously parse [this code](https://play.rust-lang.org/?gist=5cf023825e7eeffa1ff5ac42f293c60a)?

```rust
fn main() {
    match 0 {
        1 => loop {}
        _ => {}
    }
}
```

Comment (dtolnay) [reply src/expr.rs]: Oh this changed in nightly! Never mind.

File: src/main.rs (hunk 2 above)
Comment (alexcrichton) [reply src/main.rs]: Weird that was not supposed to sneak in...

File: src/expr.rs (hunk 3 above)
Comment (alexcrichton) [reply src/expr.rs]: Indeed!

**Pull request merged**